lua/nvim-tree/actions/dispatch.lua

```diff
@@ -1,10 +1,12 @@
+local api = vim.api
+
 local view = require "nvim-tree.view"
 local lib = require "nvim-tree.lib"
 
 local M = {}
 
 local Actions = {
-  close = view.close,
+  close = api.close,
```

Comment:
It looks like you're referencing the vim api instead of nvim-tree

Reply:
> I think I'm missing something fundamental here. Pressing `q` on a node results in it being opened rather than the tree closing.
> 
> nvim -nu [nvt-dev.lua](https://github.com/nvim-tree/nvim-tree.lua/files/9893107/nvt-dev.lua.txt)
> 
> Pressed q on `.luacheckrc`: ...
> 
> It looks like `handle_tree_actions` is falling through to `open_file`.

This is something i broke accidentally, and i think i need some advice on how not to do that. 

I'll explain my goal with this PR in more detail.

Nvim-tree implements the `open_on_tab` flag. What this does is ensure Nvim tree is open on all tabs IFF there is already a tree open somewhere. However, once nvim tree is open there is no way to properly close it again. This PR [closes](https://github.com/nvim-tree/nvim-tree.lua/pull/1698/files#diff-daa2785203fb370deedccd5b5c910812d0069024b341af2bb770f14e4f712fc9R203) all instances of Nvim-tree on `close`. This is only done when `open_on_tab` is set. This goes pretty far in making nvim-tree "persistent" across tabs.

In order to only trigger the new behaviour if `open_on_tab` is set. `close` needs to know whether the opt is set. Thus i create [nvim-tree.lua close](https://github.com/nvim-tree/nvim-tree.lua/pull/1698/files#diff-d2d7203cea884a09928ac1574693d436d6e3db28e62daabfa53f5f9f7740f51aR89) to be the new general interfact instead of `view.close`. 

This also means `api.tree.close` is [now](https://github.com/nvim-tree/nvim-tree.lua/pull/1698/files#diff-69640a8794ab2834e7b05cd80b8c0516749f6e893813a81b4e71117f346e809fR20). `nvim-tree.close` instead of `view.close`.

I replaced all other instances of `view.close` with `api.close`, which should have been `api.tree.close`, but when i do that i get some errors, i now see. I think `nvim-tree.close` should be used here instead. But i wanted some advice here first before i change it.